### ISSUE TITLE
No templates visible

### ISSUE THREAD

**Issue by zooba** [Microsoft/RTVS#2100]
For Dev15, templates need to be distributed already extracted and with a manifest file. VS 2015 can also include templates in the same way.

When building using the latest VSSDK (installed via Nuget), this should be done automatically.

**Comment by MikhailArkhipov** [comment]
This will have to be done in 0.4 branch.

**Comment by MikhailArkhipov** [comment]
We also have code in AddNewItem command that locates zipped templates via DTE and uses them. I am wondering if we would have to write new version for Dev15 then.

**Comment by zooba** [comment]
IIRC, we just had to remove the ".zip" from the template name and it correctly resolves to the directory.

**Comment by MikhailArkhipov** [comment]
Well we do stuff like

```
        public void AddNewItem(string templateName, string name, string extension, string destinationPath) {
            var project = GetSelectedProject<IVsHierarchy>()?.GetDTEProject();
            if (project != null) {
                DTE dte = VsAppShell.Current.GetGlobalService<DTE>();
                var solution = (Solution2)dte.Solution;

                // Construct name of the compressed template
                var compressedTemplateName = Path.ChangeExtension(templateName, "zip");
                var templatePath = Path.Combine(GetProjectItemTemplatesFolder(), compressedTemplateName);

                // We will be extracting template contents into a temp folder
                var uncompressedTemplateFolder = Path.Combine(Path.GetTempPath(), templateName);
                var uncompressedTemplateName = Path.ChangeExtension(compressedTemplateName, "vstemplate");
                var tempTemplateFile = Path.Combine(uncompressedTemplateFolder, uncompressedTemplateName);

```

**Comment by MikhailArkhipov** [comment]
So it needs to be modified for Dev15 and not to mess with unzipping.

**Comment by AlexanderSher** [comment]
Why do we need to make it in 0.4 branch? I thought Preview3 is published already.

**Comment by MikhailArkhipov** [comment]
In VS 2015 Preview 4 we still ship 0.4. We may take couple of fixes, but 0.5 is way too raw.

**Comment by AlexanderSher** [comment]
Zip files are replaced with template folders
files.swr is generated with PS script
